yaml/kubernetes/security/run-as-non-root-container-level-missing-security-context.yaml

```diff
@@ -0,0 +1,77 @@
+rules:
+- id: run-as-non-root-container-level-missing-security-context
+  patterns:
+  - pattern-inside: |
+      spec:
+        ...
+        containers:
+          ...
+        ...
+  # No runAsNonRoot in Security Context at pod level
+  - pattern-not-inside: |
+      spec:
+        ...
+        securityContext:
+          ...
+          runAsNonRoot: $VAL
+        ...
+  # Containers defined
+  - pattern-inside: |
+      spec:
+        ...
+        containers:
+          ...
+  # At least one container in this pod has defined runAsNonRoot
+  # So a decision is made to secure at container level
+  - pattern-inside: |
+      spec:
+        ...
+        containers:
+          ...
+          - name: $NAME
+            image: ...
+            ...
+            securityContext:
+              ...
+              runAsNonRoot: $VALUE
+  # Capture container image
+  - pattern: |
+          - name: $CONTAINER
+            image: $IMAGE
+            ...
+  # But missing securityContext
+  - pattern-not: |
+          - name: $CONTAINER
+            image: $IMAGE
+            ...
+            securityContext:
+              ...
+  - focus-metavariable: $IMAGE
+  fix: |
+    $IMAGE
+          securityContext:
```

Comment:
I suspect the indentation here might not work as expected if the target yaml file is not intended like our test files? :(

Reply:
Correct. For now that is not something we can do anything about with text-based autofixes.